CMakeLists.txt

```diff
@@ -222,110 +222,69 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/cogcore.pc
+        add_wayland_protocol(cogplatform-fdo CLIENT text-input-unstable-v3)
+        target_compile_definitions(cogplatform-fdo PRIVATE COG_IM_API_SUPPORTED=1)
     else ()
-        add_definitions(-DCOG_IM_API_SUPPORTED=0)
+        target_compile_definitions(cogplatform-fdo PRIVATE COG_IM_API_SUPPORTED=0)
     endif ()
 
+    # Direct video display extension.
     find_path(WPEBACKEND_FDO_HAS_VIDEO_PLANE_DISPLAY_DMABUF_EXT
         NAMES wpe/extensions/video-plane-display-dmabuf.h
         PATHS ${COGPLATFORM_FDO_DEPS_INCLUDE_DIRS}
         NO_DEFAULT_PATH
     )
-    if (WPEBACKEND_FDO_HAS_VIDEO_PLANE_DISPLAY_DMABUF_EXT)
-        set(WPEBACKEND_FDO_HAS_VIDEO_PLANE_DISPLAY_DMABUF_EXT ON)
-    else ()
-        set(WPEBACKEND_FDO_HAS_VIDEO_PLANE_DISPLAY_DMABUF_EXT OFF)
-    endif ()
 
     if (COG_WESTON_DIRECT_DISPLAY AND WPEBACKEND_FDO_HAS_VIDEO_PLANE_DISPLAY_DMABUF_EXT)
-        file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/weston)
-        pkg_check_modules(LIBWESTON REQUIRED libweston-9-protocols)
-        pkg_get_variable(LIBWESTON_PKG_DATA_DIR libweston-9-protocols "pkgdatadir")
-        find_package(WaylandScanner REQUIRED)
-        add_custom_command(
-            OUTPUT ${CMAKE_BINARY_DIR}/weston/weston-direct-display-client-protocol.h
-                   ${CMAKE_BINARY_DIR}/weston/weston-direct-display-protocol.c
-            DEPENDS ${LIBWESTON_PKG_DATA_DIR}/weston-direct-display.xml
-            COMMAND ${WAYLAND_SCANNER} client-header < ${LIBWESTON_PKG_DATA_DIR}/weston-direct-display.xml > ${CMAKE_BINARY_DIR}/weston/weston-direct-display-client-protocol.h
-            COMMAND ${WAYLAND_SCANNER} ${WAYLAND_SCANNER_CODE_ARG} < ${LIBWESTON_PKG_DATA_DIR}/weston-direct-display.xml > ${CMAKE_BINARY_DIR}/weston/weston-direct-display-protocol.c
-            VERBATIM
-        )
-
-        add_custom_command(
-            OUTPUT ${CMAKE_BINARY_DIR}/weston/weston-content-protection-client-protocol.h
-                   ${CMAKE_BINARY_DIR}/weston/weston-content-protection-protocol.c
-            DEPENDS ${LIBWESTON_PKG_DATA_DIR}/weston-content-protection.xml
-            COMMAND ${WAYLAND_SCANNER} client-header < ${LIBWESTON_PKG_DATA_DIR}/weston-content-protection.xml > ${CMAKE_BINARY_DIR}/weston/weston-content-protection-client-protocol.h
-            COMMAND ${WAYLAND_SCANNER} ${WAYLAND_SCANNER_CODE_ARG} < ${LIBWESTON_PKG_DATA_DIR}/weston-content-protection.xml > ${CMAKE_BINARY_DIR}/weston/weston-content-protection-protocol.c
-            VERBATIM
-        )
-
-        list(APPEND COGPLATFORM_FDO_SOURCES
-            ${CMAKE_BINARY_DIR}/weston/weston-direct-display-protocol.c
-            ${CMAKE_BINARY_DIR}/weston/weston-content-protection-protocol.c
-        )
-        list(APPEND COGPLATFORM_FDO_INCLUDE_DIRS ${CMAKE_BINARY_DIR}/weston)
-        set(ENABLE_WESTON_DIRECT_DISPLAY 1)
+        pkg_check_modules(LIBWESTON REQUIRED libweston-8-protocols)
+        pkg_get_variable(LIBWESTON_PKG_DATA_DIR libweston-8-protocols "pkgdatadir")
```

Comment:
Used to be `libweston-9-protocols`?

Reply:
Indeed, good catch!

I have pushed an update that brings back the :nine: